Sources/_Datadog_Private/ObjcAppLaunchHandler.m

```diff
@@ -4,60 +4,94 @@
 * Copyright 2019-2020 Datadog, Inc.
 */
 
-#import <UIKit/UIKit.h>
 #import "ObjcAppLaunchHandler.h"
+#import <sys/sysctl.h>
+#import <UIKit/UIKit.h>
+
+// `AppLaunchHandler` aims to track some times as part of the sequence described in Apple's "About the App Launch Sequence"
+// https://developer.apple.com/documentation/uikit/app_and_environment/responding_to_the_launch_of_your_app/about_the_app_launch_sequence
 
-@interface AppLaunchTimer : NSObject
-@property NSDate *frameworkLoadTime;
-// Knowing how this value is collected and that it is written only once, the use of
-// `atomic` property is enough for thread safety. It guarantees that neither partial
-// nor garbage value will be returned.
-@property (atomic) NSTimeInterval timeToApplicationBecomeActive;
+// Note that the following properties will be set early enough in the App lifecycle to avoid any race conditions without any specific synchronization means.
+static NSTimeInterval TimeToApplicationFinishedLaunching = 0.0;
+static NSTimeInterval TimeToLoadApplicationRootUI = 0.0;
+
+@interface AppLaunchHandler : NSObject
 @end
 
-@implementation AppLaunchTimer
+@implementation AppLaunchHandler {
+    NSTimeInterval _processStartTime;
+    NSTimeInterval _applicationMainEndTime;
+    NSTimeInterval _frameworkLoadTime;
+}
+
++ (void)load {
+    // This is called at the `_Datadog_Private` load time, keep the work minimal
+    NSTimeInterval now = CFAbsoluteTimeGetCurrent();
+    // Schedule a call for the first turn of run loop before the main nib/storyboard is loaded:
+    [NSRunLoop.mainRunLoop performInModes:@[NSRunLoopCommonModes] block:^{
```

Comment:
Some piece of offline feedback @buranmert shared with me:
This run loop callback is specifically there as a mean to call `markEndOfUIApplicationMain` at the right time. This is a value that we will not use yet. So maybe it would be better to take note of this solution for later and get rid of this block (i.e. just `alloc`/`init` and call `scheduleUIApplicationDidFinishLaunchingMarking` inline).

Reply:
IMO: if we don't use it → remove it as it's just a dead code. Do I miss something 🤔?